arbnode/seq_coordinator.go

```diff
@@ -671,46 +675,68 @@ func (c *SeqCoordinator) Start(ctxIn context.Context) {
 	}
 }
 
-func (c *SeqCoordinator) waitForHandoff(ctx context.Context) string {
-	var nextChosen string
+// Calls check() on retryInterval until it returns a non-default value, which is then returned.
+// On context timeout, this returns the default value.
+func waitFor[T comparable](ctx context.Context, retryInterval time.Duration, check func() T) T {
```

Comment:
seems simpler and possibly more descriptive to just have func return a boolean and wait until that's true?

Reply:
Makes sense. I've changed it to that :)